workflow/rules/fastp.smk

```diff
@@ -0,0 +1,41 @@
+# vim: syntax=python tabstop=4 expandtab
+# coding: utf-8
+
+__author__ = "Patrik Smeds"
+__copyright__ = "Copyright 2021, Patrik Smeds"
+__email__ = "patrik.smeds@scilifelab.uu.se"
+__license__ = "GPL3"
+
+
+rule fastp_pe:
+    input:
+        sample=lambda wildcards: [
+            get_fastq_file(units, wildcards, "fastq1"),
+            get_fastq_file(units, wildcards, "fastq2"),
+        ],
+    output:
+        trimmed=[
+            "prealignment/fastp_pe/{sample}_{run}_{lane}_{unit}_fastq1.fastq.gz",
+            "prealignment/fastp_pe/{sample}_{run}_{lane}_{unit}_fastq2.fastq.gz",
+        ],
+        html="prealignment/fastp_pe/{sample}_{run}_{lane}_{unit}.html",
+        json="prealignment/fastp_pes/{sample}_{run}_{lane}_{unit}.json",
+    params:
+        adapters=lambda wildcards: " --adapter_sequence {} --adapter_sequence_r2 {} ".format(
+            *get_fastq_adapter(units, wildcards).split(",")
+        ),
+        extra=config.get("fastp", {}).get("extra", ""),
+    conda:
```

Comment:
You will have to use "--use-conda" flag to aktivate conda and I guess it will not use conda if you only have "--use-singularity". But it looks like snakemake can do some fancy stuff if you combine conda and singularity flags https://snakemake.readthedocs.io/en/stable/snakefiles/deployment.html#containerization-of-conda-based-workflows

Reply:
ah, ok, then I think it doesn't matter so much. But sure, we might want to take a look at it if we want to use them together 👍